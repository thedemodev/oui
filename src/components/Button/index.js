import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Spinner from '../Spinner';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Button = ({
  ariaLabel,
  isSubmit,
  isLink,
  children,
  isActive,
  isDisabled,
  isLoading,
  loadingText,
  onBlur,
  onClick,
  onMouseDown,
  size,
  style,
  testSection,
  width = 'default',
  buttonRef,
  title,
}) => {

  const buttonClassNames = classNames(
    'oui-button', {
      [`oui-button--${style}`]: style,
      [`oui-button--${size}`]: size,
      [`oui-button--${width}`]: width,
      'is-active': isActive,
      'oui-button--loading': isLoading,
    });

  const type = isSubmit ? 'submit' : 'button';

  function handleOnClick(event) {
    if (isDisabled || isLoading) {
      return;
    }
    onClick(event);
  }

  if (isLink) {
    return (
      <div
        data-oui-component={ true }
        className={ buttonClassNames }
        disabled={ isDisabled }
        onBlur={ onBlur }
        data-test-section={ testSection }
        ref={ buttonRef }>
        { children }
      </div>
    );
  }

  return (
    <button
      data-oui-component={ true }
      className={ buttonClassNames }
      disabled={ isDisabled || isLoading }
      type={ type }
      onBlur={ onBlur }
      onClick={ handleOnClick }
      onMouseDown={ onMouseDown }
      data-test-section={ testSection }
      aria-label={ ariaLabel }
      aria-live="polite"
      title={ title }
      ref={ buttonRef }>
      { isLoading && <Spinner data-test-section="button-spinner" size="tiny"/> }
      { isLoading ? loadingText || 'Processing' : children }
    </button>
  );
};

Button.propTypes = {
  /** Describes buttons that have an icon but no text */
  ariaLabel: PropTypes.string,
  /** React ref to the underlying button component */
  buttonRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
  /** Text within the button */
  children: PropTypes.node.isRequired,
  /** Render button with active state */
  isActive: PropTypes.bool,
  /** Prevent users from interacting with the button */
  isDisabled: PropTypes.bool,
  /** Changes the button to a div for insertion within a Link component */
  isLink: PropTypes.bool,
  /** When true, adds a spinner to the button and disables the button */
  isLoading: PropTypes.bool,
  /** Make the button act as a submit button */
  isSubmit: PropTypes.bool,
  /** When the button adds a spinner, it displays this text */
  loadingText: PropTypes.string,
  /** Function that fires when the button loses focus */
  onBlur: PropTypes.func,
  /** Function that fires when the button is clicked on */
  onClick: PropTypes.func,
  /** Function that fires when the button is mouse downed */
  onMouseDown: PropTypes.func,
  /** Various height and width options */
  size: PropTypes.oneOf([
    'tiny',
    'small',
    'large',
    'narrow',
    'tight',
  ]),
  /** Various color options */
  style: PropTypes.oneOf([
    'highlight',
    'danger',
    'danger-outline',
    'outline',
    'outline-reverse',
    'plain',
    'toggle',
    'underline',
    'unstyled',
  ]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** Title of the button shown as tooltip text when the button is hovered */
  title: PropTypes.string,
  /** Various height and width options */
  width: PropTypes.oneOf([
    'default',
    'full',
  ]),
};

Button.defaultProps = {
  isLink: false,
  isLoading: false,
  isDisabled: false,
  isSubmit: false,
  loadingText: '',
  onBlur: () => {},
  onClick: () => {},
  onMouseDown: () => {},
  width: 'default',
};

Button.displayName = 'Button';

export default Button;
