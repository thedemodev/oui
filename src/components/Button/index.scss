////
/// Buttons
/// @example[html] Button colors
///   <button class="oui-button">Button</button>
///   <button class="oui-button oui-button--highlight">Button</button>
///   <button class="oui-button oui-button--danger">Button</button>
///   <button class="oui-button oui-button--danger-outline">Button</button>
///   <button class="oui-button oui-button--outline">Button</button>
///   <button class="oui-button oui-button--underline">Button <span class="oui-arrow-inline--down"></span></button>
///   <button class="oui-button oui-button--plain">Button</button>
/// @example[html] Tiny buttons
///   <button class="oui-button oui-button--tiny">Button</button>
///   <button class="oui-button oui-button--small">Button</button>
/// @example[html] Full width buttons
///   <button class="oui-button oui-button--full">Button</button>
/// @example[html] Disabled buttons
///   <button class="oui-button oui-button--disabled">Disabled with Class</button>
///   <button class="oui-button" disabled>Disabled with Attribute</button>
////

%#{$namespace}button,
.#{$namespace}button,
%oui-button,
.oui-button {
  display: inline-block;
  vertical-align: middle;
  white-space: nowrap;
  font-family: inherit;
  cursor: pointer;
  margin: 0;
  line-height: map-fetch($button, size, base, line-height);
  border-width: 1px;
  border-style: solid;
  font-size: map-fetch($button, size, base, font-size);
  font-weight: map-fetch($button, size, base, font-weight);
  user-select: none;
  border-radius: map-fetch($button, size, base, border-radius);
  height: map-fetch($button, size, base, height);
  padding: map-fetch($button, size, base, padding);
  transition: color map-fetch($transition-duration, base), background-color map-fetch($transition-duration, base), box-shadow map-fetch($transition-duration, base), border-color map-fetch($transition-duration, base);

  // Unique styles for the base, button start here.
  background-color: map-fetch($button, type, base, background);
  border-color: map-fetch($button, type, base, border);
  color: map-fetch($button, type, base, color);

  &:visited {
    color: map-fetch($button, type, base, color);
  }

  &:hover {
    color: map-fetch($button, type, base, color);
    background-color: $base-white;
  }

  &:focus {
    // Uses the same style as input hover and removes the outline while
    // remaining accessible.
    outline: none;
    border: 1px solid map-fetch($color, ui, medium);
  }

  &.is-active,
  &:active {
    background: map-fetch($button, type, base, active, background);
    border-color: map-fetch($button, type, base, active, border);
    color: map-fetch($button, type, base, active, color);
    box-shadow: map-fetch($button, type, base, active, box-shadow);
  }

  &--highlight {
    background-color: map-fetch($button, type, highlight, background);
    border-color: map-fetch($button, type, highlight, border);
    color: map-fetch($color, text, white);

    &:hover {
      background: map-fetch($button, type, highlight, hover, background);
      border-color: map-fetch($button, type, highlight, hover, border);
    }

    &:focus {
      border: 1px solid map-fetch($button, type, highlight, focus, border);
    }

    &:active {
      background: map-fetch($button, type, highlight, active, background);
      border-color: map-fetch($button, type, highlight, active, border);
      box-shadow: map-fetch($button, type, highlight, active, box-shadow);
    }

    &:visited,
    &:hover,
    &:active,
    &:focus {
      color: map-fetch($color, text, white);
    }
  }

  &--outline {
    background-color: map-fetch($button, type, outline, background);
    border-color: map-fetch($button, type, outline, border);
    color: map-fetch($button, type, outline, color);

    &:visited {
      color: map-fetch($button, type, outline, color);
    }

    &:focus {
      border: 1px solid map-fetch($button, type, outline, hover, border);
    }

    &:hover {
      background-color: map-fetch($button, type, outline, background);
      border-color: map-fetch($button, type, outline, hover, border);
      color: map-fetch($button, type, outline, hover, color);
      box-shadow: none;
    }

    &:active {
      background-color: map-fetch($button, type, outline, background);
      border-color: map-fetch($button, type, outline, active, border);
      color: map-fetch($button, type, outline, active, color);
      box-shadow: none;
    }
  }

  &--danger {
    background-color: map-fetch($button, type, danger, background);
    border-color: map-fetch($button, type, danger, border);
    color: map-fetch($color, text, white);

    &:hover {
      background: map-fetch($button, type, danger, hover, background);
      border-color: map-fetch($button, type, danger, hover, border);
    }

    &:active {
      background: map-fetch($button, type, danger, active, background);
      border-color: map-fetch($button, type, danger, active, border);
    }

    &:focus {
      border: 1px solid shade(map-fetch($button, type, danger, background), 30%);
    }

    &:visited,
    &:hover,
    &:active,
    &:focus {
      color: map-fetch($color, text, white);
    }
  }

  &--danger-outline {
    color: map-fetch($button, type, danger-outline, color);
    border-color: map-fetch($button, type, danger-outline, border);
    background-color: transparent;
    box-shadow: none;

    &:focus {
      border: 1px solid shade(map-fetch($button, type, danger, background), 30%);
    }

    &:hover {
      color: map-fetch($button, type, danger-outline, hover, color);
      border-color: map-fetch($button, type, danger-outline, hover, border);
      background-color: transparent;
      box-shadow: none;
    }

    &:active {
      border-color: map-fetch($button, type, danger-outline, active, border);
      background-color: transparent;
      color: map-fetch($button, type, danger-outline, active, color);
      box-shadow: none;
    }
  }

  &--plain,
  &--underline {
    color: map-fetch($button, type, plain, color);
    border-color: transparent;
    background-color: transparent;
    box-shadow: none;

    &:hover {
      color: map-fetch($button, type, plain, hover, color);
    }

    &:active {
      color: map-fetch($button, type, plain, active, color);
    }

    &:visited,
    &:hover,
    &:active,
    &:focus {
      border-color: transparent;
      background-color: transparent;
      box-shadow: none;
    }
  }

  &--underline {
    border-radius: 0 !important;
  }

  &--plain {
    &:focus, &:hover {
      border: 1px solid var(--light-blue-25);
    }
  }

  &--toggle {
    &:hover {
      color: map-fetch($button, type, base, color);
    }

    &:active,
    &.is-active:hover {
      color: map-fetch($button, type, base, active, color);
    }

    &:not(:hover):not(:active):not(.is-active) {
      border-color: transparent;
      background-color: transparent;
    }
  }

  &--underline {
    border-bottom: 2px solid map-fetch($button, type, underline, border);
    color: map-fetch($button, type, base, color);
    padding-left: spacer(0.5);
    padding-right: spacer(0.5);

    &:hover {
      border-bottom-width: 2px;
      border-bottom-color: map-fetch($button, type, underline, hover, border);
      color: map-fetch($button, type, underline, color);
      & .oui-arrow-inline--down {
        color: map-fetch($button, type, underline, hover, arrow-color);
      }
    }

    &:focus {
      border-bottom-width: 2px;
      border-bottom-color: map-fetch($button, type, underline, focus, border);
      & .oui-arrow-inline--down {
        color: map-fetch($button, type, underline, focus, border);
      }
    }

    &:active {
      border-bottom-width: 2px;
      border-bottom-color: map-fetch($button, type, underline, active, border);
      color: map-fetch($button, type, underline, active, color);
      & .oui-arrow-inline--down {
        color: map-fetch($button, type, underline, active, arrow-color);
      }
    }
  }

  // Button Sizes

  &--tiny {
    font-size: map-fetch($button, size, tiny, font-size);
    font-weight: map-fetch($button, size, tiny, font-weight);
    border-radius: map-fetch($button, size, tiny, border-radius);
    line-height: map-fetch($button, size, tiny, line-height);
    height: map-fetch($button, size, tiny, height);
    padding: map-fetch($button, size, tiny, padding);
  }

  &--small {
    font-size: map-fetch($button, size, small, font-size);
    font-weight: map-fetch($button, size, small, font-weight);
    border-radius: map-fetch($button, size, small, border-radius);
    line-height: map-fetch($button, size, small, line-height);
    height: map-fetch($button, size, small, height);
    padding: map-fetch($button, size, small, padding);
  }

  &--large {
    font-size: map-fetch($button, size, large, font-size);
    font-weight: map-fetch($button, size, large, font-weight);
    border-radius: map-fetch($button, size, large, border-radius);
    line-height: map-fetch($button, size, large, line-height);
    height: map-fetch($button, size, large, height);
    padding: map-fetch($button, size, large, padding);
  }

  &--narrow {
    padding-left: spacer(1);
    padding-right: spacer(1);
  }

  &--tight {
    padding-left: spacer(0.5);
    padding-right: spacer(0.5);
  }

  &--full {
    width: 100%;
    text-align: center;
  }

  // With Icon
  // Some extra help is needed to get the icon and text to line up.

  &--icon {
    line-height: 1.4;

    .#{$namespace}icon {
      vertical-align: sub;
    }
  }

  &--unstyled {
    // scss-lint:disable ImportantRule
    background: inherit !important;
    border: none !important;
    border-radius: inherit !important;
    box-shadow: inherit !important;
    color: inherit !important;
    font-size: inherit !important;
    font-weight: inherit !important;
    height: inherit !important;
    line-height: 1 !important;
    margin: 0 !important;
    padding: 0 !important;
    text-shadow: inherit !important;
  }

  // Disabled State
  // Used either an attribute or a class, depending on the use case.

  &[disabled],
  &--disabled,
  &--disabled:visited,
  &--disabled:hover,
  &--disabled:active,
  &--disabled:focus {
    cursor: default;
    box-shadow: none;
    background-color: map-fetch($button, type, disabled, background);
    border-color: map-fetch($button, type, disabled, border);
    color: map-fetch($button, type, disabled, color);
    box-shadow: none;

    &.#{$namespace}button--plain,
    &.#{$namespace}button--toggle,
    &.oui-button--plain,
    &.oui-button--toggle {
      background-color: transparent;
      border-color: transparent;
    }

    &.#{$namespace}button--outline-reverse,
    &.oui-button--outline-reverse {
      background-color: transparent;
    }
  }

  &--loading {
    .oui-spinner {
      margin-bottom: spacer(-0.5);
      margin-right: spacer(0.5);
    }
  }
}
