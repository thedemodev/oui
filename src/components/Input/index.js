import React from 'react';
import Label from '../Label';
import classNames from 'classnames';
import PropTypes from 'prop-types';

/**
 * Generates an `input` element (optionally wrapped in a label) and accepts
 * most of the common input types.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

class Input extends React.Component {
  blur() {
    if (this._input) {
      this._input.blur();
    }
  }

  renderInput({
    isFilter,
    displayError,
    type,
    value,
    defaultValue,
    placeholder,
    isRequired,
    isReadOnly,
    isDisabled,
    onInput,
    onChange,
    onBlur,
    onKeyDown,
    onFocus,
    min,
    max,
    step,
    maxLength,
    testSection,
    focus,
    textAlign,
    id }) {

    let hasAlignStyle = false;
    if (textAlign) {
      hasAlignStyle = true;
    }

    let classes = classNames(
      'oui-text-input',
      {'oui-text-input--read-only': isReadOnly},
      {'oui-text-input--search': isFilter},
      {'oui-form-bad-news': displayError},
      {[`text--${textAlign}`]: hasAlignStyle}
    );

    return (
      /* eslint-disable react/jsx-no-bind */
      <input
        data-oui-component={ true }
        className={ classes }
        id={ id }
        ref={ (c) => { this._input = c; } }
        type={ type }
        value={ value }
        defaultValue={ defaultValue }
        placeholder={ placeholder }
        required={ isRequired }
        readOnly={ isReadOnly }
        disabled={ isDisabled }
        onInput={ onInput }
        onChange={ onChange }
        onBlur={ onBlur }
        onKeyDown={ onKeyDown }
        onFocus={ onFocus }
        min={ min }
        max={ max }
        step={ step }
        { ...(typeof maxLength === 'undefined' ? {} : { maxLength }) }
        data-test-section={ testSection }
        autoFocus={ focus }
      />
      /* eslint-enable */
    );
  }

  renderNote({note, testSection}) {
    return (<div className='oui-form-note' data-test-section={ testSection && testSection + '-note' }>{note}</div>);
  }

  render() {
    const {
      label,
      displayError,
      testSection,
      isOptional,
      note,
      isRequired,
      id,
    } = this.props;

    if (label) {
      return (
        <div
          data-oui-component={ true }
          className={ classNames({'oui-form-bad-news': displayError}) }>
          <Label
            testSection={ testSection && testSection + '-label' }
            isRequired={ isRequired }
            isOptional={ isOptional }
            inputId={ id }>
            { label }
          </Label>
          { this.renderInput(this.props) }
          { note && this.renderNote(this.props)}
        </div>
      );
    }

    if (note) {
      return [
        this.renderInput(this.props),
        this.renderNote(this.props),
      ];
    }

    return this.renderInput(this.props);

  }
}

Input.propTypes = {
  /** The default value of the input used on initial render */
  defaultValue: PropTypes.string,
  /** Includes search icon if true */
  displayError: PropTypes.bool,
  /** Id of the input to properly associate with the input's label */
  id: PropTypes.string,
  /** Prevents input from being modified and appears disabled */
  isDisabled: PropTypes.bool,
  /** Includes error if true */
  isFilter: PropTypes.bool,
  /** Adds an optional label if there is a label provided
   *  @param {Object} props Object of props
   *  @returns {Error} Error or null
   */
  isOptional: function verifyIsOptionalProp(props) {
    if (props.isOptional && !props.label) {
      return new Error('Must include a value for the label prop to use the isOptional prop');
    }
    return null;
  },
  /** Prevents input from being modified but doesn't appear disabled */
  isReadOnly: PropTypes.bool,
  /** Includes required asterisk label if true
   *  @param {Object} props Object of props
   *  @returns {Error} Error or null
   */
  isRequired: function verifyIsRequiredProp(props) {
    if (props.isRequired && !props.label) {
      return new Error('Must include a value for the label prop to use the isRequired prop');
    }
    return null;
  },
  /** Text that describes the input */
  label: PropTypes.string,
  /**
   * Max value for the `input`. Should be used only when `type` is `number`.
   */
  max: PropTypes.number,
  /**
   * Max length of the input value. Should be used only when type is 'text',
   * 'email', 'search', 'password', 'tel', or 'url'.
   */
  maxLength: PropTypes.number,
  /**
   * Min value for the `input`. Should be used only when `type` is `number`.
   */
  min: PropTypes.number,
  /** Form note for the input */
  note: PropTypes.string,
  /**
   * Function that fires when the input loses focus. It fires regardless of
   * whether the value has changed.
  */
  onBlur: PropTypes.func,
  /** Function that fires when the input loses focus after the value changes */
  onChange: PropTypes.func,
  /** Function that fires when the input gains focus */
  onFocus: PropTypes.func,
  /** Function that fires on keypress */
  onInput: PropTypes.func,
  /** Function that fires when a key is pressed down */
  onKeyDown: PropTypes.func,
  /** Input placeholder text */
  placeholder: PropTypes.string,
  /** Input step value */
  step: PropTypes.string,
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** Align text inside input. Default is left. */
  textAlign: PropTypes.oneOf(['left', 'right']),
  /** Supported input types */
  type: PropTypes.oneOf([
    'text',
    'password',
    'date',
    'number',
    'email',
    'url',
    'search',
    'tel',
    'time',
  ]).isRequired,
  /** Text within the input */
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

Input.defaultProps = {
  note: null,
  isRequired: false,
};

export default Input;
