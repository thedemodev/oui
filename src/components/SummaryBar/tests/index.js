import React from 'react';
import ExampleComponent from '../index';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Input from '../../Input';
import Button from '../../Button';

describe('components/ExampleComponent', function() {
  let component;
  let fnMock = jest.fn(event => { return event; });

  beforeEach(function() {
    component = mount(
      <ExampleComponent
        aString={ 'aStringValue' }
        aNumber={ 5 }
        aBoolean={ true }
        aFunction={ fnMock }
        anArray={ [ 1, 2, 3 ] }
        anObject={{ initialValue: 'INITIAL_VALUE' }}>
        { ({
          aString,
          aNumber,
          aBoolean,
          aFunction,
          anArray,
          anObject,
          handleInputUpdate,
          name,
        }) => (
          <React.Fragment>
            <Input type="text" onInput={ handleInputUpdate } testSection="nameInput"/>
            <div data-test-section="inputNameValue">{ name }</div>
            <Button style="outline" onClick={ aFunction } testSection="pressMe">Press Me!</Button>
            <div data-test-section="aString">{ String(aString) }</div>
            <div data-test-section="aNumber">{ String(aNumber) }</div>
            <div data-test-section="aBoolean">{ String(aBoolean) }</div>
            <div data-test-section="anArray">{ JSON.stringify(anArray) }</div>
            <div data-test-section="anObject">{ JSON.stringify(anObject) }</div>
          </React.Fragment>
        ) }
      </ExampleComponent>
    );
  });

  it('should update "aStringValue" prop', function() {
    expect(component.find('[data-test-section="aString"]').text()).toBe('aStringValue (with updates)');
  });

  it('should update "aNumber" prop', function() {
    expect(component.find('[data-test-section="aNumber"]').text()).toBe('6');
  });

  it('should update "aBoolean" prop', function() {
    expect(component.find('[data-test-section="aBoolean"]').text()).toBe('false');
  });

  it('should update "anArray" prop', function() {
    expect(component.find('[data-test-section="anArray"]').text()).toBe('[1,2,3,4,5,6]');
  });

  it('should update "anObject" prop', function() {
    expect(component.find('[data-test-section="anObject"]').text()).toBe('{\"initialValue\":\"INITIAL_VALUE\",\"newKey\":true}');
  });

  it('should update ExampleComponent state and return name to children when a user types their name in an input', function() {
    expect(component.state()).toMatchObject({ name: ''});
    component.find('[data-test-section="nameInput"]').simulate('input', { target: { value: 'Johnny' } });
    expect(component.state()).toMatchObject({ name: 'Johnny'});
    expect(component.find('[data-test-section="inputNameValue"]').text()).toBe('Johnny');
  });

  it('should call aFunction with a string argument of "updated" when the Button is pressed', function() {
    component.find('[data-test-section="pressMe"]').simulate('click');
    expect(fnMock.mock.calls[0][0]).toBe('updated');
  });

  it('should match the Jest Snapshot', function() {
    expect(mountToJson(component)).toMatchSnapshot();
  });
});
