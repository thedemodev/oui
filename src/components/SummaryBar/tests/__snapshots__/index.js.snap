// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ExampleComponent should match the Jest Snapshot 1`] = `
<ExampleComponent
  aBoolean={true}
  aFunction={
    [MockFunction] {
      "calls": Array [
        Array [
          "updated",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": "updated",
        },
      ],
    }
  }
  aNumber={5}
  aString="aStringValue"
  anArray={
    Array [
      1,
      2,
      3,
    ]
  }
  anObject={
    Object {
      "initialValue": "INITIAL_VALUE",
    }
  }
>
  <div
    className="oui-example-component"
  >
    <Input
      isRequired={false}
      note={null}
      onInput={[Function]}
      testSection="nameInput"
      type="text"
    >
      <input
        className="oui-text-input"
        data-oui-component={true}
        data-test-section="nameInput"
        onInput={[Function]}
        required={false}
        type="text"
      />
    </Input>
    <div
      data-test-section="inputNameValue"
    />
    <Button
      isDisabled={false}
      isLink={false}
      isLoading={false}
      isSubmit={false}
      loadingText=""
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      style="outline"
      testSection="pressMe"
      width="default"
    >
      <button
        aria-live="polite"
        className="oui-button oui-button--outline oui-button--default"
        data-oui-component={true}
        data-test-section="pressMe"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        type="button"
      >
        Press Me!
      </button>
    </Button>
    <div
      data-test-section="aString"
    >
      aStringValue (with updates)
    </div>
    <div
      data-test-section="aNumber"
    >
      6
    </div>
    <div
      data-test-section="aBoolean"
    >
      false
    </div>
    <div
      data-test-section="anArray"
    >
      [1,2,3,4,5,6]
    </div>
    <div
      data-test-section="anObject"
    >
      {"initialValue":"INITIAL_VALUE","newKey":true}
    </div>
  </div>
</ExampleComponent>
`;
